### PR TITLE
Update VSC's DeletionPolicy to Retain when it is set to Delete.

### PR DESCRIPTION
**Pull request by blackpiglet**
Fix issue https://github.com/vmware-tanzu/velero/issues/4945

Signed-off-by: Xun Jiang <jxun@vmware.com>

### CODE REVIEW

File: internal/restore/volumesnapshot_action.go [hunk 1]
```diff
@@ -93,6 +93,12 @@ func (p *VolumeSnapshotRestoreItemAction) Execute(input *velero.RestoreItemActio
 			deletionPolicy = string(snapshotv1api.VolumeSnapshotContentRetain)
 		}
 
+		if deletionPolicy == string(snapshotv1api.VolumeSnapshotContentDelete) {
```
Comment (reasonerjt) [inline internal/restore/volumesnapshot_action.go:96]: why not just ignore the annotation and set the deletion policy always to retain, and print a explicit message.

**Comment by reasonerjt** [comment]
Could you also update the doc, to point this out in `Implementation choices`:
https://github.com/vmware-tanzu/velero/blob/4cca3996f507a170561f4683f2c47427e7b541e3/site/content/docs/main/csi.md#L36

File: internal/restore/volumesnapshot_action.go (hunk 1 above)
Comment (blackpiglet) [reply internal/restore/volumesnapshot_action.go:96]: Reasonable. My thought is limited the log print numbers.

**Comment by reasonerjt** [comment]
1.  Please also revise the commit msg to explicitly point out that this change is for restore process.
2.  Please double check if the change to handle vsc resource is necessary.

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**